### PR TITLE
[AppService] functionapp: Fixed issue where app setting was being incorrectly set for linux custom images

### LINKED ISSUE #14795: [AppService] Linux Dedicated with custom image shouldn't set FUNCTIONS_WORKER_RUNTIME
When creating a Linux Dedicated functionapp with a custom image, the cli is setting FUNCTIONS_WORKER_RUNTIME to dotnet. This config should be set to null seeing as the custom image doesn't have to be dotnet.

### PR DESCRIPTION
**Pull request by gzuber**
**Description**  

Resolves #14795 

This fixes and issue where if a customer is using a linux custom container image for their function, the `FUNCTIONS_WORKER_RUNTIME` app setting was being set incorrectly.

**Testing Guide**  

Fixed an existing test to make sure this scenario is validated.

```
azdev test test_acr_create_function_app
```

**History Notes**  

[AppService] functionapp create: Fixed issue where app setting was being incorrectly set for linux custom images

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AppService

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/recordings/test_acr_create_function_app.yaml [hunk 1]
```diff
@@ -63,24 +63,24 @@ interactions:
...
+      string: '{"sku":{"name":"Basic","tier":"Basic"},"type":"Microsoft.ContainerRegistry/registries","id":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.ContainerRegistry/registries/functionappacrtest000004","name":"functionappacrtest000004","location":"brazilsouth","tags":{},"systemData":{"createdBy":"grzuber@microsoft.com","createdByType":"User","createdAt":"2020-09-14T13:18:11.2342671+00:00","lastModifiedBy":"grzuber@microsoft.com","lastModifiedByType":"User","lastModifiedAt":"2020-09-14T13:18:11.2342671+00:00"},"properties":{"loginServer":"functionappacrtest000004.azurecr.io","creationDate":"2020-09-14T13:18:11.2342671Z","provisioningState":"Succeeded","adminUserEnabled":true,"policies":{"quarantinePolicy":{"status":"disabled"},"trustPolicy":{"type":"Notary","status":"disabled"},"retentionPolicy":{"days":7,"lastUpdatedTime":"2020-09-14T13:18:15.0681241+00:00","status":"disabled"}},"encryption":{"status":"disabled"},"dataEndpointEnabled":false,"dataEndpointHostNames":[],"privateEndpointConnections":[],"publicNetworkAccess":"Enabled"}}'
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/appservice/tests/latest/recordings/test_acr_create_function_app.yaml:74]: why change the ACR test?

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 2]
```diff
@@ -2816,11 +2817,16 @@ def create_function(cmd, resource_group_name, name, storage_account, plan=None,
                 site_config.app_settings.append(NameValuePair(name='WEBSITES_ENABLE_APP_SERVICE_STORAGE',
                                                               value='false'))
                 site_config.linux_fx_version = _format_fx_version(deployment_container_image_name)
+
+                # clear all runtime specific configs and settings
+                site_config_json = {KEYS.USE_32_BIT_WORKER_PROC: False}
+                app_settings_json = {}
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:2823]: this seems dangrous, will user's app_settings be cleared also?

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/recordings/test_acr_create_function_app.yaml (hunk 1 above)
Comment (gzuber) [reply src/azure-cli/azure/cli/command_modules/appservice/tests/latest/recordings/test_acr_create_function_app.yaml:74]: The ACR test was creating a linux custom image function app, but mistakenly assuming that the `FUNCTIONS_WORKER_RUNTIME` would be set to `node` when it shouldn't be set at all. By changing the test, we're now confirming that that app setting won't be set. The only thing I changed was what value the test expected `FUNCTIONS_WORKER_RUNTIME` to be.

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py (hunk 2 above)
Comment (gzuber) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py:2823]: Yeah sorry this is a little misleading. I'm not clearing anything from the functionapp definition (what we use in the request to make the function app).

These two variables (`site_config_json` and `app_settings_json`) only contain json from the stacks API, and represent what app settings and site configs we'll be setting  on the functionapp definition _later_ -- [here specifically](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/appservice/custom.py#L2826-L2833). Since linux custom image apps shouldn't be using the stacks api (they're providing their own image that contains these app settings instead of using our image and stacks api settings), we're just pre-emptively clearing these settings before they overwrite what's in the functionapp definition.

In a nutshell we're saying -- "Since you're providing your own image and settings, we'll go ahead and clear the settings we would normally set before we set them, especially since we don't even know what runtime your custom image is using."

**Review by qwordy — APPROVED** [review]
